### DISCUSSION TITLE
How are contributions computed with the explain class

### DISCUSSION THREAD

**Discussion by arthurserres** [pyg-team/pytorch_geometric#8595] [General]
Hello all, 

I try to understand the way the contributions are computed by the explain library. 

I use Captum explainer and in particular Integrated Gradient method. I read the paper explaining the integrated gradient method so the method by itself is clear. However I looked at the source code as well at the documentation but I do not manage to understand how each node feature abd edge contribution are computed, may someone help on that ?

Thanks a lot :)

**Comment by rusty1s** [comment]
For integration with GNNs, we wrap the utilized GNN into a parent module, which expects two inputs: The node feature matrix, and a soft edge mask which is multiplied within message passing to every message. These are the two inputs that Captum will generate attributions for. Hope this answers your question.

**Reply by arthurserres** [reply]
Thanks for the clarification, I think it helps me quite  alot but I oppened another discussion in order to precise my question: [here](https://github.com/pyg-team/pytorch_geometric/discussions/8612) 
I would really appreciate if you manage to give an answer :)
